README.md

```diff
@@ -29,24 +29,77 @@ the [ngx_lua module](http://wiki.nginx.org/HttpLuaModule), [LuaJIT 2.0](http://l
 [api-gateway-hmac](https://github.com/adobe-apiplatform/api-gateway-hmac) module, and
 [lua-resty-http](https://github.com/pintsized/lua-resty-http) module.
 
-### AWS V4 Signature
-This library supports the latest AWS V4 signature which means you can use any of the latest AWS APIs without any problem.
+### AWS Credentials Provider
+Requests to AWS Services must supply valid credentials and this library provides a few credentials providers for signing AWS Requests.
+`aws_credentials` config option specifies which provider to use. 
+
+If no `aws_credentials` is provided then the library will try to guess one:
```

Comment:
If no `aws_credentials` is provided then the library will try to find one using the following order:
1. 
2.